src/pages/cli/restapi/override.mdx

```diff
@@ -152,3 +152,132 @@ user who is invoking the
+        },
+        authorizerResultTtlInSeconds: 0
+      },
+    },
+  });
+```
+
+As API Gateway needs permission to invoke the Authorizer lambda function, add resource based policy to the function using following code:
+
+```ts
+// Adding Resource Based policy to Lambda Authorizer function
+resources.addCfnResource(
+    {
+      type: "AWS::Lambda::Permission",
+      properties: {
+            Action: "lambda:InvokeFunction",
+            FunctionName: {Ref: functionArnParameter},
+            Principal: "apigateway.amazonaws.com",
+            SourceArn:{
+                "Fn::Join": [
+                  "",
+                  [
+                    "arn:aws:execute-api:",
+                    {
+                      "Ref": "AWS::Region"
+                    },
+                    ":",
+                    {
+                      "Ref": "AWS::AccountId"
+                    },
+                    ":",
+                    {
+                      "Ref": "<your-API-resource-name>"
+                    },
+                    "/*/*"
+                  ]
+                ]
+              }
+            }
+    },"LambdaAuthorizerResourceBasedPolicy");
+
+```
+<Callout warning>
+
+Make sure to replace `<your-api-resource-name>` with the name of your Rest API resource.
+This is the name of the folder in `amplify/backend/api` that was created when
+you added an Rest API resource to your Amplify project.
+
+</Callout>
+
+Finally, update the security methods for all of the paths in your REST API to use this new Lambda authorizer. You can also add the Authorization header as a parameter on incoming requests for these paths as a place for users to provide their Auth token. 
+
+```ts
+for (const path in resources.restApi.body.paths) {
+    // Add the Authorization header as a parameter to requests
+    resources.restApi.addPropertyOverride(
+      `Body.paths.${path}.x-amazon-apigateway-any-method.parameters`,
+      [
+        ...resources.restApi.body.paths[path]["x-amazon-apigateway-any-method"]
+          .parameters,
+        {
+          name: "Authorization",
+          in: "header",
+          required: false,
+          type: "string",
+        },
+      ]
+    );
+    // Use your new Lambda authorizer for security
+    resources.restApi.addPropertyOverride(
+      `Body.paths.${path}.x-amazon-apigateway-any-method.security`,
+      [ { Lambda: [], }, ]
+    );
+  }
+```
+
+
+You can also configure Request Based Lambda Authorizer referring the OpenAPI extension, [`x-amazon-apigateway-authorizer`](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-authorizer.html).
+Additionally you can add more advanced logic to only use the Lambda authorizer
+with some paths or methods.
+
+
+When performing requests to your REST API, make sure to add the Authorization header with an token required by Lambda Authorizer Function.
```

Comment:
Show a code snippet here with for the client-side execution

Reply:
Hello,
Referring the documentation for cognito authorizer, I could see client code is mentioned in lib section. Hence added the client code for lambda authorizer in lib section -> Define authorization rules.